From d0f510befa2ae708dbe4091d845764cd0e7dd6d8 Mon Sep 17 00:00:00 2001
From: Salvo Giangreco <giangrecosalvo9@gmail.com>
Date: Mon, 6 Oct 2025 14:23:19 +0200
Subject: [PATCH] Introduce GamesPropsUtils

UN1CA: this patch is not complete! Please check unica/mods/gamesprops/customize.sh for more info.
---
 .../io/mesalabs/unica/GamesPropsUtils.smali   | 979 ++++++++++++++++++
 1 file changed, 979 insertions(+)
 create mode 100644 smali_classes6/io/mesalabs/unica/GamesPropsUtils.smali

diff --git a/smali_classes6/io/mesalabs/unica/GamesPropsUtils.smali b/smali_classes6/io/mesalabs/unica/GamesPropsUtils.smali
new file mode 100644
index 00000000..a3dc5958
--- /dev/null
+++ b/smali_classes6/io/mesalabs/unica/GamesPropsUtils.smali
@@ -0,0 +1,979 @@
+.class public final Lio/mesalabs/unica/GamesPropsUtils;
+.super Ljava/lang/Object;
+.source "GamesPropsUtils.java"
+
+
+# static fields
+.field private static final blacklist DEBUG:Z = false
+
+.field private static final blacklist TAG:Ljava/lang/String; = "GamesPropsUtils"
+
+.field private static final blacklist packagesToChange:Ljava/util/Map;
+    .annotation system Ldalvik/annotation/Signature;
+        value = {
+            "Ljava/util/Map<",
+            "Ljava/lang/String;",
+            "[",
+            "Ljava/lang/String;",
+            ">;"
+        }
+    .end annotation
+.end field
+
+.field private static final blacklist propsToChange:Ljava/util/Map;
+    .annotation system Ldalvik/annotation/Signature;
+        value = {
+            "Ljava/util/Map<",
+            "Ljava/lang/String;",
+            "Ljava/util/Map<",
+            "Ljava/lang/String;",
+            "Ljava/lang/Object;",
+            ">;>;"
+        }
+    .end annotation
+.end field
+
+
+# direct methods
+.method static constructor blacklist <clinit>()V
+    .locals 13
+
+    new-instance v0, Ljava/util/HashMap;
+
+    invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
+
+    sput-object v0, Lio/mesalabs/unica/GamesPropsUtils;->propsToChange:Ljava/util/Map;
+
+    new-instance v1, Ljava/util/HashMap;
+
+    invoke-direct {v1}, Ljava/util/HashMap;-><init>()V
+
+    sput-object v1, Lio/mesalabs/unica/GamesPropsUtils;->packagesToChange:Ljava/util/Map;
+
+    invoke-static {}, Lio/mesalabs/unica/GamesPropsUtils;->createBS4Props()Ljava/util/Map;
+
+    move-result-object v2
+
+    const-string v3, "BS4"
+
+    invoke-interface {v0, v3, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    const/4 v2, 0x1
+
+    new-array v4, v2, [Ljava/lang/String;
+
+    const-string v5, "com.proximabeta.mf.uamo"
+
+    const/4 v6, 0x0
+
+    aput-object v5, v4, v6
+
+    invoke-interface {v1, v3, v4}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    invoke-static {}, Lio/mesalabs/unica/GamesPropsUtils;->createF5Props()Ljava/util/Map;
+
+    move-result-object v3
+
+    const-string v4, "F5"
+
+    invoke-interface {v0, v4, v3}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    const/4 v3, 0x3
+
+    new-array v5, v3, [Ljava/lang/String;
+
+    const-string v7, "com.dts.freefiremax"
+
+    aput-object v7, v5, v6
+
+    const-string v7, "com.dts.freefireth"
+
+    aput-object v7, v5, v2
+
+    const-string v7, "com.mobile.legends"
+
+    const/4 v8, 0x2
+
+    aput-object v7, v5, v8
+
+    invoke-interface {v1, v4, v5}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    invoke-static {}, Lio/mesalabs/unica/GamesPropsUtils;->createMI11TPProps()Ljava/util/Map;
+
+    move-result-object v4
+
+    const-string v5, "MI11TP"
+
+    invoke-interface {v0, v5, v4}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    const/4 v4, 0x4
+
+    new-array v7, v4, [Ljava/lang/String;
+
+    const-string v9, "com.ea.gp.apexlegendsmobilefps"
+
+    aput-object v9, v7, v6
+
+    const-string v9, "com.levelinfinite.hotta.gp"
+
+    aput-object v9, v7, v2
+
+    const-string v9, "com.supercell.clashofclans"
+
+    aput-object v9, v7, v8
+
+    const-string v9, "com.vng.mlbbvn"
+
+    aput-object v9, v7, v3
+
+    invoke-interface {v1, v5, v7}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    invoke-static {}, Lio/mesalabs/unica/GamesPropsUtils;->createMI13PProps()Ljava/util/Map;
+
+    move-result-object v5
+
+    const-string v7, "MI13P"
+
+    invoke-interface {v0, v7, v5}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    new-array v5, v8, [Ljava/lang/String;
+
+    const-string v9, "com.levelinfinite.sgameGlobal"
+
+    aput-object v9, v5, v6
+
+    const-string v9, "com.tencent.tmgp.sgame"
+
+    aput-object v9, v5, v2
+
+    invoke-interface {v1, v7, v5}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    invoke-static {}, Lio/mesalabs/unica/GamesPropsUtils;->createOP8PProps()Ljava/util/Map;
+
+    move-result-object v5
+
+    const-string v7, "OP8P"
+
+    invoke-interface {v0, v7, v5}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    const/4 v5, 0x7
+
+    new-array v9, v5, [Ljava/lang/String;
+
+    const-string v10, "com.netease.lztgglobal"
+
+    aput-object v10, v9, v6
+
+    const-string v10, "com.riotgames.league.wildrift"
+
+    aput-object v10, v9, v2
+
+    const-string v10, "com.riotgames.league.wildrifttw"
+
+    aput-object v10, v9, v8
+
+    const-string v10, "com.riotgames.league.wildriftvn"
+
+    aput-object v10, v9, v3
+
+    const-string v10, "com.riotgames.league.teamfighttactics"
+
+    aput-object v10, v9, v4
+
+    const-string v10, "com.riotgames.league.teamfighttacticstw"
+
+    const/4 v11, 0x5
+
+    aput-object v10, v9, v11
+
+    const-string v10, "com.riotgames.league.teamfighttacticsvn"
+
+    const/4 v12, 0x6
+
+    aput-object v10, v9, v12
+
+    invoke-interface {v1, v7, v9}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    invoke-static {}, Lio/mesalabs/unica/GamesPropsUtils;->createOP9PProps()Ljava/util/Map;
+
+    move-result-object v7
+
+    const-string v9, "OP9P"
+
+    invoke-interface {v0, v9, v7}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    new-array v7, v3, [Ljava/lang/String;
+
+    const-string v10, "com.epicgames.fortnite"
+
+    aput-object v10, v7, v6
+
+    const-string v10, "com.epicgames.portal"
+
+    aput-object v10, v7, v2
+
+    const-string v10, "com.tencent.lolm"
+
+    aput-object v10, v7, v8
+
+    invoke-interface {v1, v9, v7}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    invoke-static {}, Lio/mesalabs/unica/GamesPropsUtils;->createROG6Props()Ljava/util/Map;
+
+    move-result-object v7
+
+    const-string v9, "ROG6"
+
+    invoke-interface {v0, v9, v7}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    new-array v7, v11, [Ljava/lang/String;
+
+    const-string v10, "com.ea.gp.fifamobile"
+
+    aput-object v10, v7, v6
+
+    const-string v10, "com.gameloft.android.ANMP.GloftA9HM"
+
+    aput-object v10, v7, v2
+
+    const-string v10, "com.madfingergames.legends"
+
+    aput-object v10, v7, v8
+
+    const-string v10, "com.pearlabyss.blackdesertm.gl"
+
+    aput-object v10, v7, v3
+
+    const-string v10, "com.pearlabyss.blackdesertm"
+
+    aput-object v10, v7, v4
+
+    invoke-interface {v1, v9, v7}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    const-string v7, "ROG6D"
+
+    invoke-static {}, Lio/mesalabs/unica/GamesPropsUtils;->createROG6DProps()Ljava/util/Map;
+
+    move-result-object v10
+
+    invoke-interface {v0, v7, v10}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    new-array v7, v2, [Ljava/lang/String;
+
+    const-string v10, "com.proxima.dfm"
+
+    aput-object v10, v7, v6
+
+    invoke-interface {v1, v9, v7}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    invoke-static {}, Lio/mesalabs/unica/GamesPropsUtils;->createS24UProps()Ljava/util/Map;
+
+    move-result-object v7
+
+    const-string v9, "S24U"
+
+    invoke-interface {v0, v9, v7}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    new-array v5, v5, [Ljava/lang/String;
+
+    const-string v7, "com.blizzard.diablo.immortal"
+
+    aput-object v7, v5, v6
+
+    const-string v7, "com.pubg.imobile"
+
+    aput-object v7, v5, v2
+
+    const-string v7, "com.pubg.krmobile"
+
+    aput-object v7, v5, v8
+
+    const-string v7, "com.rekoo.pubgm"
+
+    aput-object v7, v5, v3
+
+    const-string v7, "com.tencent.ig"
+
+    aput-object v7, v5, v4
+
+    const-string v7, "com.tencent.tmgp.pubgmhd"
+
+    aput-object v7, v5, v11
+
+    const-string v7, "com.vng.pubgmobile"
+
+    aput-object v7, v5, v12
+
+    invoke-interface {v1, v9, v5}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    invoke-static {}, Lio/mesalabs/unica/GamesPropsUtils;->createLenovoY700Props()Ljava/util/Map;
+
+    move-result-object v5
+
+    const-string v7, "LenovoY700"
+
+    invoke-interface {v0, v7, v5}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    new-array v0, v4, [Ljava/lang/String;
+
+    const-string v4, "com.activision.callofduty.shooter"
+
+    aput-object v4, v0, v6
+
+    const-string v4, "com.garena.game.codm"
+
+    aput-object v4, v0, v2
+
+    const-string v2, "com.tencent.tmgp.kr.codm"
+
+    aput-object v2, v0, v8
+
+    const-string v2, "com.vng.codmvn"
+
+    aput-object v2, v0, v3
+
+    invoke-interface {v1, v7, v0}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    return-void
+.end method
+
+.method private constructor blacklist <init>()V
+    .locals 0
+
+    invoke-direct {p0}, Ljava/lang/Object;-><init>()V
+
+    return-void
+.end method
+
+.method private static blacklist createBS4Props()Ljava/util/Map;
+    .locals 4
+    .annotation system Ldalvik/annotation/Signature;
+        value = {
+            "()",
+            "Ljava/util/Map<",
+            "Ljava/lang/String;",
+            "Ljava/lang/Object;",
+            ">;"
+        }
+    .end annotation
+
+    new-instance v0, Ljava/util/HashMap;
+
+    invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
+
+    const-string v1, "BRAND"
+
+    const-string v2, "blackshark"
+
+    invoke-interface {v0, v1, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    const-string v1, "DEVICE"
+
+    const-string v3, "penrose"
+
+    invoke-interface {v0, v1, v3}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    const-string v1, "MANUFACTURER"
+
+    invoke-interface {v0, v1, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    const-string v1, "MODEL"
+
+    const-string v2, "SHARK PRS-A0"
+
+    invoke-interface {v0, v1, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    return-object v0
+.end method
+
+.method private static blacklist createF5Props()Ljava/util/Map;
+    .locals 3
+    .annotation system Ldalvik/annotation/Signature;
+        value = {
+            "()",
+            "Ljava/util/Map<",
+            "Ljava/lang/String;",
+            "Ljava/lang/Object;",
+            ">;"
+        }
+    .end annotation
+
+    new-instance v0, Ljava/util/HashMap;
+
+    invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
+
+    const-string v1, "BRAND"
+
+    const-string v2, "POCO"
+
+    invoke-interface {v0, v1, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    const-string v1, "DEVICE"
+
+    const-string v2, "marble"
+
+    invoke-interface {v0, v1, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    const-string v1, "MANUFACTURER"
+
+    const-string v2, "Xiaomi"
+
+    invoke-interface {v0, v1, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    const-string v1, "MODEL"
+
+    const-string v2, "23049PCD8G"
+
+    invoke-interface {v0, v1, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    return-object v0
+.end method
+
+.method private static blacklist createLenovoY700Props()Ljava/util/Map;
+    .locals 4
+    .annotation system Ldalvik/annotation/Signature;
+        value = {
+            "()",
+            "Ljava/util/Map<",
+            "Ljava/lang/String;",
+            "Ljava/lang/Object;",
+            ">;"
+        }
+    .end annotation
+
+    new-instance v0, Ljava/util/HashMap;
+
+    invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
+
+    const-string v1, "BRAND"
+
+    const-string v2, "Lenovo"
+
+    invoke-interface {v0, v1, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    const-string v1, "DEVICE"
+
+    const-string v3, "TB-9707F"
+
+    invoke-interface {v0, v1, v3}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    const-string v1, "MANUFACTURER"
+
+    invoke-interface {v0, v1, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    const-string v1, "MODEL"
+
+    const-string v2, "Lenovo TB-9707F"
+
+    invoke-interface {v0, v1, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    return-object v0
+.end method
+
+.method private static blacklist createMI11TPProps()Ljava/util/Map;
+    .locals 4
+    .annotation system Ldalvik/annotation/Signature;
+        value = {
+            "()",
+            "Ljava/util/Map<",
+            "Ljava/lang/String;",
+            "Ljava/lang/Object;",
+            ">;"
+        }
+    .end annotation
+
+    new-instance v0, Ljava/util/HashMap;
+
+    invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
+
+    const-string v1, "BRAND"
+
+    const-string v2, "Xiaomi"
+
+    invoke-interface {v0, v1, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    const-string v1, "DEVICE"
+
+    const-string v3, "vili"
+
+    invoke-interface {v0, v1, v3}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    const-string v1, "MANUFACTURER"
+
+    invoke-interface {v0, v1, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    const-string v1, "MODEL"
+
+    const-string v2, "2107113SI"
+
+    invoke-interface {v0, v1, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    return-object v0
+.end method
+
+.method private static blacklist createMI13PProps()Ljava/util/Map;
+    .locals 4
+    .annotation system Ldalvik/annotation/Signature;
+        value = {
+            "()",
+            "Ljava/util/Map<",
+            "Ljava/lang/String;",
+            "Ljava/lang/Object;",
+            ">;"
+        }
+    .end annotation
+
+    new-instance v0, Ljava/util/HashMap;
+
+    invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
+
+    const-string v1, "BRAND"
+
+    const-string v2, "Xiaomi"
+
+    invoke-interface {v0, v1, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    const-string v1, "DEVICE"
+
+    const-string v3, "nuwa"
+
+    invoke-interface {v0, v1, v3}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    const-string v1, "MANUFACTURER"
+
+    invoke-interface {v0, v1, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    const-string v1, "MODEL"
+
+    const-string v2, "2210132C"
+
+    invoke-interface {v0, v1, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    return-object v0
+.end method
+
+.method private static blacklist createOP8PProps()Ljava/util/Map;
+    .locals 4
+    .annotation system Ldalvik/annotation/Signature;
+        value = {
+            "()",
+            "Ljava/util/Map<",
+            "Ljava/lang/String;",
+            "Ljava/lang/Object;",
+            ">;"
+        }
+    .end annotation
+
+    new-instance v0, Ljava/util/HashMap;
+
+    invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
+
+    const-string v1, "BRAND"
+
+    const-string v2, "OnePlus"
+
+    invoke-interface {v0, v1, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    const-string v1, "DEVICE"
+
+    const-string v3, "OnePlus8Pro"
+
+    invoke-interface {v0, v1, v3}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    const-string v1, "MANUFACTURER"
+
+    invoke-interface {v0, v1, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    const-string v1, "MODEL"
+
+    const-string v2, "IN2020"
+
+    invoke-interface {v0, v1, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    return-object v0
+.end method
+
+.method private static blacklist createOP9PProps()Ljava/util/Map;
+    .locals 4
+    .annotation system Ldalvik/annotation/Signature;
+        value = {
+            "()",
+            "Ljava/util/Map<",
+            "Ljava/lang/String;",
+            "Ljava/lang/Object;",
+            ">;"
+        }
+    .end annotation
+
+    new-instance v0, Ljava/util/HashMap;
+
+    invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
+
+    const-string v1, "BRAND"
+
+    const-string v2, "OnePlus"
+
+    invoke-interface {v0, v1, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    const-string v1, "DEVICE"
+
+    const-string v3, "OnePlus9Pro"
+
+    invoke-interface {v0, v1, v3}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    const-string v1, "MANUFACTURER"
+
+    invoke-interface {v0, v1, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    const-string v1, "MODEL"
+
+    const-string v2, "LE2123"
+
+    invoke-interface {v0, v1, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    return-object v0
+.end method
+
+.method private static blacklist createROG6DProps()Ljava/util/Map;
+    .locals 4
+    .annotation system Ldalvik/annotation/Signature;
+        value = {
+            "()",
+            "Ljava/util/Map<",
+            "Ljava/lang/String;",
+            "Ljava/lang/Object;",
+            ">;"
+        }
+    .end annotation
+
+    new-instance v0, Ljava/util/HashMap;
+
+    invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
+
+    const-string v1, "BRAND"
+
+    const-string v2, "asus"
+
+    invoke-interface {v0, v1, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    const-string v1, "DEVICE"
+
+    const-string v3, "AI2203_C"
+
+    invoke-interface {v0, v1, v3}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    const-string v1, "MANUFACTURER"
+
+    invoke-interface {v0, v1, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    const-string v1, "MODEL"
+
+    const-string v2, "ASUS_AI2203_C"
+
+    invoke-interface {v0, v1, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    return-object v0
+.end method
+
+.method private static blacklist createROG6Props()Ljava/util/Map;
+    .locals 4
+    .annotation system Ldalvik/annotation/Signature;
+        value = {
+            "()",
+            "Ljava/util/Map<",
+            "Ljava/lang/String;",
+            "Ljava/lang/Object;",
+            ">;"
+        }
+    .end annotation
+
+    new-instance v0, Ljava/util/HashMap;
+
+    invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
+
+    const-string v1, "BRAND"
+
+    const-string v2, "asus"
+
+    invoke-interface {v0, v1, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    const-string v1, "DEVICE"
+
+    const-string v3, "AI2201"
+
+    invoke-interface {v0, v1, v3}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    const-string v1, "MANUFACTURER"
+
+    invoke-interface {v0, v1, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    const-string v1, "MODEL"
+
+    const-string v2, "ASUS_AI2201"
+
+    invoke-interface {v0, v1, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    return-object v0
+.end method
+
+.method private static blacklist createS24UProps()Ljava/util/Map;
+    .locals 4
+    .annotation system Ldalvik/annotation/Signature;
+        value = {
+            "()",
+            "Ljava/util/Map<",
+            "Ljava/lang/String;",
+            "Ljava/lang/Object;",
+            ">;"
+        }
+    .end annotation
+
+    new-instance v0, Ljava/util/HashMap;
+
+    invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
+
+    const-string v1, "BRAND"
+
+    const-string v2, "samsung"
+
+    invoke-interface {v0, v1, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    const-string v1, "DEVICE"
+
+    const-string v3, "e3q"
+
+    invoke-interface {v0, v1, v3}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    const-string v1, "MANUFACTURER"
+
+    invoke-interface {v0, v1, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    const-string v1, "MODEL"
+
+    const-string v2, "SM-S928B"
+
+    invoke-interface {v0, v1, v2}, Ljava/util/Map;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
+
+    return-object v0
+.end method
+
+.method private static blacklist dlog(Ljava/lang/String;)V
+    .locals 0
+
+    return-void
+.end method
+
+.method private static blacklist setPropValue(Ljava/lang/String;Ljava/lang/Object;)V
+    .locals 2
+
+    const-string v0, "Defining prop "
+
+    :try_start_0
+    new-instance v1, Ljava/lang/StringBuilder;
+
+    invoke-direct {v1, v0}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
+
+    invoke-virtual {v1, p0}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+
+    move-result-object v0
+
+    const-string v1, " to "
+
+    invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+
+    move-result-object v0
+
+    invoke-virtual {p1}, Ljava/lang/Object;->toString()Ljava/lang/String;
+
+    move-result-object v1
+
+    invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+
+    move-result-object v0
+
+    invoke-virtual {v0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
+
+    move-result-object v0
+
+    invoke-static {v0}, Lio/mesalabs/unica/GamesPropsUtils;->dlog(Ljava/lang/String;)V
+
+    const-class v0, Landroid/os/Build;
+
+    invoke-virtual {v0, p0}, Ljava/lang/Class;->getDeclaredField(Ljava/lang/String;)Ljava/lang/reflect/Field;
+
+    move-result-object v0
+
+    const/4 v1, 0x1
+
+    invoke-virtual {v0, v1}, Ljava/lang/reflect/Field;->setAccessible(Z)V
+
+    const/4 v1, 0x0
+
+    invoke-virtual {v0, v1, p1}, Ljava/lang/reflect/Field;->set(Ljava/lang/Object;Ljava/lang/Object;)V
+
+    const/4 p1, 0x0
+
+    invoke-virtual {v0, p1}, Ljava/lang/reflect/Field;->setAccessible(Z)V
+    :try_end_0
+    .catch Ljava/lang/NoSuchFieldException; {:try_start_0 .. :try_end_0} :catch_0
+    .catch Ljava/lang/IllegalAccessException; {:try_start_0 .. :try_end_0} :catch_0
+
+    return-void
+
+    :catch_0
+    move-exception p1
+
+    new-instance v0, Ljava/lang/StringBuilder;
+
+    const-string v1, "Failed to set prop "
+
+    invoke-direct {v0, v1}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
+
+    invoke-virtual {v0, p0}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+
+    move-result-object p0
+
+    invoke-virtual {p0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
+
+    move-result-object p0
+
+    const-string v0, "GamesPropsUtils"
+
+    invoke-static {v0, p0, p1}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
+
+    return-void
+.end method
+
+.method public static blacklist setProps(Landroid/content/Context;)V
+    .locals 3
+
+    invoke-virtual {p0}, Landroid/content/Context;->getPackageName()Ljava/lang/String;
+
+    move-result-object p0
+
+    invoke-static {p0}, Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z
+
+    move-result v0
+
+    if-eqz v0, :cond_0
+
+    goto :goto_1
+
+    :cond_0
+    const-string v0, "persist.sys.unica.gamehooks"
+
+    const/4 v1, 0x1
+
+    invoke-static {v0, v1}, Landroid/os/SemSystemProperties;->getBoolean(Ljava/lang/String;Z)Z
+
+    move-result v0
+
+    if-nez v0, :cond_1
+
+    goto :goto_1
+
+    :cond_1
+    sget-object v0, Lio/mesalabs/unica/GamesPropsUtils;->packagesToChange:Ljava/util/Map;
+
+    invoke-interface {v0}, Ljava/util/Map;->keySet()Ljava/util/Set;
+
+    move-result-object v0
+
+    invoke-interface {v0}, Ljava/util/Set;->iterator()Ljava/util/Iterator;
+
+    move-result-object v0
+
+    :cond_2
+    invoke-interface {v0}, Ljava/util/Iterator;->hasNext()Z
+
+    move-result v1
+
+    if-eqz v1, :cond_3
+
+    invoke-interface {v0}, Ljava/util/Iterator;->next()Ljava/lang/Object;
+
+    move-result-object v1
+
+    check-cast v1, Ljava/lang/String;
+
+    sget-object v2, Lio/mesalabs/unica/GamesPropsUtils;->packagesToChange:Ljava/util/Map;
+
+    invoke-interface {v2, v1}, Ljava/util/Map;->get(Ljava/lang/Object;)Ljava/lang/Object;
+
+    move-result-object v2
+
+    check-cast v2, [Ljava/lang/String;
+
+    invoke-static {v2}, Ljava/util/Arrays;->asList([Ljava/lang/Object;)Ljava/util/List;
+
+    move-result-object v2
+
+    invoke-interface {v2, p0}, Ljava/util/List;->contains(Ljava/lang/Object;)Z
+
+    move-result v2
+
+    if-eqz v2, :cond_2
+
+    new-instance v0, Ljava/lang/StringBuilder;
+
+    const-string v2, "Defining props for: "
+
+    invoke-direct {v0, v2}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
+
+    invoke-virtual {v0, p0}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+
+    move-result-object p0
+
+    invoke-virtual {p0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
+
+    move-result-object p0
+
+    invoke-static {p0}, Lio/mesalabs/unica/GamesPropsUtils;->dlog(Ljava/lang/String;)V
+
+    sget-object p0, Lio/mesalabs/unica/GamesPropsUtils;->propsToChange:Ljava/util/Map;
+
+    invoke-interface {p0, v1}, Ljava/util/Map;->get(Ljava/lang/Object;)Ljava/lang/Object;
+
+    move-result-object p0
+
+    check-cast p0, Ljava/util/Map;
+
+    invoke-interface {p0}, Ljava/util/Map;->entrySet()Ljava/util/Set;
+
+    move-result-object p0
+
+    invoke-interface {p0}, Ljava/util/Set;->iterator()Ljava/util/Iterator;
+
+    move-result-object p0
+
+    :goto_0
+    invoke-interface {p0}, Ljava/util/Iterator;->hasNext()Z
+
+    move-result v0
+
+    if-eqz v0, :cond_3
+
+    invoke-interface {p0}, Ljava/util/Iterator;->next()Ljava/lang/Object;
+
+    move-result-object v0
+
+    check-cast v0, Ljava/util/Map$Entry;
+
+    invoke-interface {v0}, Ljava/util/Map$Entry;->getKey()Ljava/lang/Object;
+
+    move-result-object v1
+
+    check-cast v1, Ljava/lang/String;
+
+    invoke-interface {v0}, Ljava/util/Map$Entry;->getValue()Ljava/lang/Object;
+
+    move-result-object v0
+
+    invoke-static {v1, v0}, Lio/mesalabs/unica/GamesPropsUtils;->setPropValue(Ljava/lang/String;Ljava/lang/Object;)V
+
+    goto :goto_0
+
+    :cond_3
+    :goto_1
+    return-void
+.end method
-- 
2.51.0

